nixos/modules/services/misc/pufferpanel.nix

```diff
@@ -0,0 +1,401 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.pufferpanel;
+
+  ipwrap = address:
+    if lib.hasInfix ":" address
+    then "[" + address + "]"
+    else address;
+
+  joinHostPort = host: port:
+    "${ipwrap host}:${toString port}";
+
+  boolToStringOrNull = x:
+    if x != null then
+      lib.boolToString x
+    else
+      null;
+
+  options = lib.filterAttrs (_: v: v != null) (
+    {
+      PUFFER_WEB_HOST = joinHostPort cfg.host cfg.port;
+      PUFFER_PANEL_ENABLE = lib.boolToString cfg.panel.enable;
+      PUFFER_DAEMON_ENABLE = lib.boolToString cfg.daemon.enable;
+    }
+    //
+    lib.optionalAttrs cfg.panel.enable {
+      PUFFER_PANEL_REGISTRATIONENABLED = boolToStringOrNull cfg.panel.registrationEnabled;
+    }
+    //
+    lib.optionalAttrs cfg.daemon.enable {
+      PUFFER_DAEMON_SFTP_HOST = joinHostPort cfg.daemon.sftp.host cfg.daemon.sftp.port;
+      PUFFER_DAEMON_AUTH_URL = cfg.daemon.auth.url;
+      PUFFER_DAEMON_AUTH_CLIENTID = cfg.daemon.auth.clientId;
+    }
+  );
+
+  secrets = lib.filterAttrs (_: v: v != null) (
+    lib.optionalAttrs cfg.daemon.enable {
+      PUFFER_DAEMON_AUTH_CLIENTSECRET = cfg.daemon.auth.clientSecretFile;
+    }
+  );
+
+  directories = {
+    PUFFER_LOGS = "$LOGS_DIRECTORY";
+  } // lib.optionalAttrs cfg.daemon.enable {
+    PUFFER_DAEMON_DATA_CACHE = "$CACHE_DIRECTORY";
+    PUFFER_DAEMON_DATA_SERVERS = "$STATE_DIRECTORY/servers";
+    PUFFER_DAEMON_DATA_BINARIES = "$STATE_DIRECTORY/binaries";
+  };
+
+  compatEnv = lib.optionalAttrs (cfg.compat.enable) {
+    LD_LIBRARY_PATH = "/lib";
+  };
+
+  compatPath = pkgs.buildEnv {
+    name = "pufferpanel-compat";
+    paths =
+      lib.optional (cfg.compat.enable && cfg.compat.dynamicLinker != null) cfg.compat.dynamicLinker
+      ++
+      cfg.extraPackages;
+    pathsToLink =
+      [ "/bin" "/lib" ]
+      ++
+      lib.optional pkgs.stdenv.is64bit "/lib64";
+  };
+
+  compatLayer = lib.optionals (cfg.compat.enable) (
+    let bind = path: "-${compatPath}${path}:${path}"; in
+    map bind compatPath.pathsToLink
+  );
+
+  stdenvDynamicLinker = pkgs.runCommand "pufferpanel-linker" { } ''
+    mkdir -p $out/lib
+    ln -s -t $out/lib "$(cat ${pkgs.stdenv.cc}/nix-support/dynamic-linker)"
+    ${lib.optionalString pkgs.stdenv.is64bit ''
+      ln -s $out/lib $out/lib64
+    ''}
+  '';
+in
+{
+  options.services.pufferpanel = {
+    enable = lib.mkEnableOption (lib.mdDoc ''
+      PufferPanel game management server.
```

Comment:
This will be prefixed and suffixed with some text. See https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/web-servers/nginx/default.nix#L463 which becomes ``Whether to enable Nginx Web Server.``

Reply:
Done, changed that to `lib.mkOption`.